### PR TITLE
build: add support for NetBSD in depends

### PR DESCRIPTION
**Pull request by fanquake**
Similar to #23948. Doesn't build the Qt package; I haven't looked at doing that yet, but have an assumption that it's going to fail out of the box similar to the FreeBSD build.

Guix Build:
```bash
```

### CODE REVIEW

**Comment by kristapsk** [comment]
If some NetBSD bitcoiner is looking at this, would be cool to help wih review / test of this - https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1066. None of JM developers are using NetBSD, so we can't test this ourselves.

**Comment by fanquake** [comment]
Rebased and added Guix build to PR description.

**Comment by hebasto** [comment]
Tested d25f3d5c0d49474321fa6ad7fac1610998a5d3a1 in fresh installation of NetBSD 9.2.

Packages that were required to build depends:
```
# pkgin install autoconf automake bash gcc8 gmake libtool pkg-config
```

Building depends:
```
$ gmake -C depends NO_QT=1 CC=/usr/pkg/gcc8/bin/gcc CXX=/usr/pkg/gcc8/bin/g++
```

Building Bitcoin Core:
```
$ ./autogen.sh
$ CONFIG_SITE=$PWD/depends/x86_64-unknown-netbsd9.2/share/config.site ./configure
$ gmake
...
```

Failure with Boost.Process:
VirtualBox_NetBSD 9_14_02_2022_00_24_01

**Comment by hebasto** [comment]
> Failure with Boost.Process

Tested #24254 with this PR, and it fixes this error.

File: depends/hosts/netbsd.mk [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+x86_64_netbsd_NM=nm
+x86_64_netbsd_STRIP=strip
+else
+i686_netbsd_CC=$(default_host_CC) -m32
+i686_netbsd_CXX=$(default_host_CXX) -m32
+x86_64_netbsd_CC=$(default_host_CC) -m64
+x86_64_netbsd_CXX=$(default_host_CXX) -m64
+endif
+
+netbsd_cmake_system=NetBSD
```
Comment (hebasto) [inline depends/hosts/netbsd.mk:31]: FWIW, CMake [docs](https://cmake.org/cmake/help/v3.23/manual/cmake-toolchains.7.html) say nothing about cross compiling for NetBSD, and about setting of the `CMAKE_SYSTEM_NAME` macro for such a case.

**Review by hebasto — APPROVED** [review]
ACK d25f3d5c0d49474321fa6ad7fac1610998a5d3a1

**Review by hebasto — APPROVED** [review]
re-ACK 28efee627d4b53f1a774abdc26e5050dd32465fd

Rebasing verified with
```
$ git range-diff master d25f3d5c0d49474321fa6ad7fac1610998a5d3a1 28efee627d4b53f1a774abdc26e5050dd32465fd
```

File: depends/hosts/netbsd.mk [hunk 2]
```diff
@@ -0,0 +1,31 @@
+netbsd_CFLAGS=-pipe
+netbsd_CFLAGS_CXXFLAGS=$(netbsd_CFLAGS)
```
Comment (theuni) [inline depends/hosts/netbsd.mk:2]: Should be `netbsd_CXXFLAGS`, no?

Looks like maybe the same search/replace problem with the freebsd host too?

**Review by theuni — COMMENTED** [review]
I suppose this is just for convenience, and not intended to be supported in any way? If so, Concept ACK

The additions look good to me other than the wonky CXXFLAGS.

My only concern is that since this isn't being tested, what happens if/when builds stop working and nobody notices or bothers to fix. Maybe explicitly describe it as unsupported somewhere? That way at least people would be hesitant to rely on it. Same goes for freebsd, I suppose.

**Comment by fanquake** [comment]
> I suppose this is just for convenience, and not intended to be supported in any way? 

Yes. The intention is minimal support, but enough that depends is usable on *BSD platforms. I will be testing builds irregularly, however given that we aren't supporting Qt, and the continued reduction in dependencies, I also don't expect things too break too often.

File: depends/Makefile [hunk 3]
```diff
@@ -89,6 +90,7 @@ full_host_os:=$(subst $(host_arch)-$(host_vendor)-,,$(canonical_host))
 host_os:=$(findstring linux,$(full_host_os))
 host_os+=$(findstring darwin,$(full_host_os))
 host_os+=$(findstring freebsd,$(full_host_os))
+host_os+=$(findstring netbsd,$(full_build_os))
```
Comment (theuni) [inline depends/Makefile:93]: `full_host_os` :)

Comment (fanquake) [reply depends/Makefile]: Addressed.

Comment (theuni) [reply depends/Makefile]: Does it still build after this change? I guess you've been testing in a VM where the build and host are the same?

Comment (fanquake) [reply depends/Makefile]: > Does it still build after this change?

Yes.

> I guess you've been testing in a VM where the build and host are the same?

I haven't been using a VM. I've been testing using NetBSD 9.99 running on aarch64 hardware.

**Review by theuni — APPROVED** [review]
ACK bcd36e14f0a2b89445f1436094de40696d6ad76c

**Pull request merged**